hearpreprocess/runner.py

```diff
@@ -127,6 +127,28 @@ def run(
             task_config["tmp_dir"] = tmp_dir
             task_config["mode"] = task_mode
             del task_config["modes"]
+
+            # The `splits` key has to be initialised outside the pipeline,
+            # since the splits are used in defining the required tasks
+            # for example, in requires method for ResampleSubcorpuses
+            if "split_mode" not in task_config:
+                raise ValueError("split_mode is a required config for all tasks")
+
+            if task_config["split_mode"] == "trainvaltest":
+                # Dataset will be partitioned into train/validation/test splits
+                task_config["splits"] = pipeline.SPLITS
+            elif task_config["split_mode"] in ["new_split_kfold", "presplit_kfold"]:
+                # Dataset will be partitioned in k-folds, either using predefined folds
+                # or with using folds defined in the pipeline
+                n_folds = task_config["nfolds"]
+                assert isinstance(n_folds, int)
+                task_config["splits"] = ["fold{:02d}".format(i) for i in range(n_folds)]
```

Comment:
I really don't love this idea of overwriting task_config with undocumented stuff that's not in the schema. How can we avoid that?

Reply:
I mean, can't we make a method here or something? Overwriting the config is really kinda fragile and confusing

Reply:
`splits` makes it simple to define the tasks. Like there are some tasks at a split level and the split key helps